torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3444,7 +3444,8 @@ def _post_backward_hook(
 
             # Wait for all ops in the current stream to finish before
             # reduce-scattering the gradient
-            # TODO (awgu): This current stream is actually the unshard stream!
+            # NOTE: The current stream may be the computation stream *or* the
+            # unshard stream, depending on the calling context.
             self._streams["post_backward"].wait_stream(torch.cuda.current_stream())
```

Comment:
wondering whether we can print 'torch.cuda.current_stream()' right at the beginning of post_backward_hook() and right at the end of pre_backward_hook() next time when we debug this?